solidity/test/bridge/Bridge.test.ts

```diff
@@ -920,108 +1205,445 @@ describe("Bridge", () => {
+                              )
+                            ).to.be.revertedWith("Deposit already swept")
+                          })
+                        }
+                      )
+
+                      context("when the single input is an unknown", () => {
+                        const data: DepositSweepTestData = SingleP2SHDeposit
+                        // Take wallet public key hash from first deposit. All
+                        // deposits in same sweep batch should have the same value
+                        // of that field.
+                        const { walletPubKeyHash } = data.deposits[0].reveal
+
+                        before(async () => {
+                          await createSnapshot()
+
+                          // Simulate the wallet is an Live one and is known in the system.
+                          await bridge.setWallet(walletPubKeyHash, {
+                            ...walletDraft,
+                            state: walletState.Live,
+                          })
+
+                          // Necessary to pass the proof validation.
+                          relay.getPrevEpochDifficulty.returns(
+                            data.chainDifficulty
+                          )
+                          relay.getCurrentEpochDifficulty.returns(
+                            data.chainDifficulty
+                          )
+                        })
+
+                        after(async () => {
+                          await restoreSnapshot()
+                        })
+
+                        it("should revert", async () => {
+                          // Try to sweep a deposit which was not revealed before and
+                          // is unknown from system's point of view.
+                          await expect(
+                            bridge.submitDepositSweepProof(
+                              data.sweepTx,
+                              data.sweepProof,
+                              NO_MAIN_UTXO,
+                              ethers.constants.AddressZero
+                            )
+                          ).to.be.revertedWith("Unknown input type")
+                        })
+                      })
+                    })
+
+                    // Since P2SH vs P2WSH path has been already checked in the scenario
+                    // "when there is only one input", we no longer differentiate deposits
+                    // using that criterion during "when there are multiple inputs" scenario.
+                    context("when there are multiple inputs", () => {
+                      context(
+                        "when input vector consists only of revealed unswept deposits and the expected main UTXO",
+                        () => {
+                          let tx: ContractTransaction
+                          const previousData: DepositSweepTestData =
+                            MultipleDepositsNoMainUtxo
```

Comment:
Why do we need `previousData`? To create `dataWithVault` we copy `data` and in the previous version we had only `const data: DepositSweepTestData = MultipleDepositsNoMainUtxo`.

Reply:
Because this is the case with multiple inputs where one input is the main UTXO. `previousData` is used to perform the first sweep and set the expected main UTXO`.